client/authorization/components/userControls.js

```diff
@@ -1,30 +1,23 @@
 import React from 'react'
-import PropTypes from 'prop-types'
-import { connect } from 'react-redux'
+import {useDispatch, useSelector} from 'react-redux'
 import * as actions from '../actions'
 import SignInDialog from './signInDialog'
 import classNames from "./scss/button.module.scss";
 
 const UserControls = (props) => {
-    const { authorized, logout} = props;
+    const dispatch = useDispatch();
+    const authorized = useSelector(state => state.application.authorized);
     if (!authorized) {
         return <SignInDialog/>
     }
 
     const {appbar__button} = classNames;
 
     return (
-        <button onClick={logout} className={appbar__button}>
+        <button onClick={() => dispatch(actions.logout())} className={appbar__button}>
```

Comment:
avoid using inline functions. Better to define handleClick function wit useCallback hook in that situation ( think why:) )